presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -584,7 +576,8 @@ ObjectMetadata getS3ObjectMetadata(Path path)
         }
         catch (Exception e) {
```

Comment:
Again, this wasn't part of your change but since we are toughing this code... what are your thoughts on not catching Exception here and instead catching  only Runtime & Checked exceptions? I ask mostly because of the possibility that in the future this could catch Interrupted Exception and as currently constructed it would not properly reset the interrupt flag when it propagates.

This seems like a common pattern throughout this class, mostly curious your thoughts on this since it may not be practical to change it throughout.

Reply:
Right above on L573 we handle `InterruptedException` separately. We do the same for the other places where we catch `Exception`.

Reply:
Ah, I totally missed that in the code folding.